pandas/core/indexes/base.py

```diff
@@ -1385,6 +1398,12 @@ def set_names(self, names, level=None, inplace: bool = False):
             idx = self
         else:
             idx = self._shallow_copy()
+
+        if isinstance(self, ABCMultiIndex) and is_dict_like(names):
+            level = Index(self.names).get_indexer_for(names)
```

Comment:
can you not just iterate over these to grab the levels? (and use get_level_values)?

Reply:
Yep, thx. Thats way better.

List comprehensions would be an option too, but would have to loop twice. Would prefer ist this way